### PR TITLE
ci: add & configure release-please

### PR DESCRIPTION
**Pull request by thor**
Adds [release-please](https://github.com/googleapis/release-please) by configuring [the release-please GitHub Action](https://github.com/google-github-actions/release-please-action).

The action provides a continually up-to-date PR that can be merged when the `main` branch is ready to be released as a new version. Through conventional commit messages it'll determine what the next version number is.
The [release-please explains what this does and how it is intended to be used](https://github.com/googleapis/release-please/blob/main/README.md#how-should-i-write-my-commits).

### CODE REVIEW

**Comment by thor** [comment]
@blt950 note that the errors are expected as I'm configuring the action and the manifest in one PR. I could split them out into separate ones though, if you'd like 🟢 first, but I've tested it locally and will fix-forward in the event of anything... strange.

**Comment by blt950** [comment]
@thor I've taken a brief look. I've never used this before, so bear with me. I understand that this CI will make all commits into a `main` into a pull request. This pull request will collect all commits to create a changelog and suggest(?) a version number bump that is commited when PR is merged. We'll still need to manually click "release" by merging the PR.

Changed we need to do in our pipeline and habits:
- Commits need to be written in a very specific way
- We don't need to write changelog manually anymore (though we still will do it for the #status-update channel in VATSCA, because it usually is simplified into a non-techy wording and explanations).

Some control questions:
- What is the advantage of doing it this way?
- What are the disadvantages of using this?
- Are we overcompliacting something that is not a challenge today?
- What do we do with commits that are not well written? E.g. merged from a external PR, a hasty push, just forgetting the new formating etc.
- Is this in line with our "there might be a another web team in future that is not as techy as we are and will be able to cope"-mantra?

**Comment by blt950** [comment]
Thanks for the info, let me think about this next time I do some fixes in CC so I've the right mindset. I'll follow up with a decision.

**Comment by thor** [comment]
> Thanks for the info, let me think about this next time I do some fixes in CC so I've the right mindset. I'll follow up with a decision.

Bump. We can always close this for the time being and leave the branch present. FWIW, besides the initial noise in the pull requests, it's possible to test and revert this if you'd like, or test it in a fork.

**Comment by blt950** [comment]
Keep it open. I'm slowly trying to work on my habits to write "correct" commits for this, once I'm confident we can merge this in :)

**Comment by blt950** [comment]
@thor Just follow up question:
- As I understand it'll create a PR based on the commits in `main`, but where does it pull into? A new version like branch? 
- Since it creates an PR, it'll be possible to use this without command line, correct?
   - If for some reason we disagree that next update is a major instead of a minor. Can this be solved without command line?

**Comment by thor** [comment]
> @thor Just follow up question:

@blt950 No worries.

>     * As I understand it'll create a PR based on the commits in `main`, but where does it pull into? A new version like branch?

That's right. It pulls it into its own "release-please--<something>" branch, from which it creates the pull request.
In other words, you can try it out and all that it'll do is create (and update) the pull request.

>     * Since it creates an PR, it'll be possible to use this without command line, correct?

Affirmative.
       
>       * If for some reason we disagree that next update is a major instead of a minor. Can this be solved without command line?

Affirmative. You can [alter already-merged changes](https://github.com/googleapis/release-please#how-can-i-fix-release-notes), or you [can override the version manually](https://github.com/googleapis/release-please#how-do-i-change-the-version-number).

**Review by blt950 — APPROVED** [review]
Behind the #643 and #638 have merged, cleared line up and merge this as well, behind.

File: .github/workflows/release-please.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+name: release-please
+
+jobs:
+  release-please:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: google-github-actions/release-please-action@v3
+        with:
+          command: manifest
```
Comment (thor) [inline .github/workflows/release-please.yaml:19]:
```suggestion
        with:
          release-type: php
```

File: .github/workflows/release-please.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+on:
+  push:
+    branches:
+      - main
+      - ci/release-please
```
Comment (thor) [inline .github/workflows/release-please.yaml:5]:
```suggestion

```

File: .release-please-manifest.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  ".": "4.0.5"
```
Comment (thor) [inline .release-please-manifest.json:2]:
```suggestion
  ".": "4.2.3"
```

File: .github/workflows/release-please.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+name: release-please
+
+jobs:
+  release-please:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: google-github-actions/release-please-action@v3
+        with:
+          release-type: php
```
Comment (thor) [inline .github/workflows/release-please.yaml:18]:
```suggestion
```

File: .github/workflows/release-please.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  contents: write
+  pull-requests: write
+
+name: release-please
+
+jobs:
+  release-please:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: google-github-actions/release-please-action@v3
```
Comment (thor) [inline .github/workflows/release-please.yaml:16]:
```suggestion
      - uses: google-github-actions/release-please-action@v3

```

File: .github/workflows/release-please.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+on:
+  push:
+    branches:
+      - main
```
Comment (thor) [inline .github/workflows/release-please.yaml:4]:
```suggestion
      - main
  workflow_dispatch:
```

File: .github/workflows/release-please.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
```
Comment (thor) [inline .github/workflows/release-please.yaml:4]:
```suggestion
      - main
      - ci/release-please

```

File: .github/workflows/release-please.yaml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  contents: write
+  pull-requests: write
+
+name: release-please
+
+jobs:
+  release-please:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: google-github-actions/release-please-action@v3
```
Comment (thor) [inline .github/workflows/release-please.yaml:19]:
```suggestion
      - uses: google-github-actions/release-please-action@v3
    with:
      release-type: php
```

File: .github/workflows/release-please.yaml [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    with:
+      release-type: php
```
Comment (thor) [inline .github/workflows/release-please.yaml:21]:
```suggestion
        with:
          release-type: php

```

**Pull request merged**